### DISCUSSION TITLE
Has DUCKDB_TYPE_JSON been removed from the C API?

### DISCUSSION THREAD

**Discussion by marcboeker** [duckdb/duckdb#6288] [General]
Hi,

I'm wondering why `DUCKDB_TYPE_JSON` has been removed from the C API in the 0.7.0 release as it was still [available in 0.6.1](	https://github.com/duckdb/duckdb/blob/919cad22e8090087ae33625661f26a5fc78d188b/src/include/duckdb.h#L123)?

Our tests are [failing](https://github.com/marcboeker/go-duckdb/actions/runs/4177556580/jobs/7235271381) because of the missing JSON type at the moment.

Thanks!

**Comment by hannes** [comment]
@lnkuiper can say more, but I believe it has been removed

**Comment by lnkuiper** [comment]
Hi @marcboeker , the `LogicalTypeId:::JSON` has been removed. Instead, we moved this type to the JSON extension, so it is no longer part of the DuckDB core. Now, it is `LogicalTypeId::VARCHAR` with the `"JSON"` alias, and it is only available if the extension is installed.

**Reply by marcboeker** [reply]
Thanks @lnkuiper for the clarification. I've now removed the `JSON` type. Were there any other changes in regards to DuckDB types? It seems that the `MAP` type had some changes too, as our [`MAP` tests](https://github.com/marcboeker/go-duckdb/actions/runs/4187233038/jobs/7256798484) are failing now.

**Reply by Mytherin** [reply]
The map has been inverted to be stored more efficiently - see https://github.com/duckdb/duckdb/pull/5768
